source/Mouse.cpp

```diff
@@ -0,0 +1,39 @@
+#include "Mouse.hpp"
+#include "UIManager.hpp"
+
+Mouse Mouse::mouseInstance;
+
+Mouse &Mouse::getMouse(void) {
+
+	return mouseInstance;
+}
+
+Mouse::Mouse(void)
+	: Input<OIS::Mouse, OIS::MouseListener>()
+{}
+
+bool Mouse::mouseMoved(OIS::MouseEvent const &) {
+
+	return true;
+}
+
+bool Mouse::mousePressed(OIS::MouseEvent const &event, OIS::MouseButtonID id) {
+
+	OIS::MouseState const &state = const_cast<OIS::MouseState const &>(event.state);
```

Comment:
[Click here](http://code.joyridelabs.de/ois_api/OISMouse_8h_source.html) !
If you look at the class MouseEvent, its member is a const MouseState.